util/pkg/vfs/s3context.go

```diff
@@ -40,16 +40,21 @@ var (
 	s3UrlRegexp = regexp.MustCompile(`s3([-.](?P<region>\w{2}-\w+-\d{1})|[-.](?P<bucket>[\w.\-\_]+)|)?.amazonaws.com(.cn)?(?P<path>.*)?`)
 )
 
+type S3Location struct {
```

Comment:
I'm wondering if there's a better name for this type - since "location" doesn't capture the encryption info it contains. Maybe something like S3BucketDetails?